components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSCarFileMigrationTool.java

```diff
@@ -0,0 +1,328 @@
+/**
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * <p>
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.dashboards.migrationtool;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.log4j.BasicConfigurator;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.*;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+
+public class DSCarFileMigrationTool extends DSMigrationTool {
+    private static final Log log = LogFactory.getLog(DSCarFileMigrationTool.class);
+    private static final String ARTIFACT_XML = "artifact.xml";
+    private static final String GADGET_ARTIFACT = "dashboards/gadget";
+    private static final String DASHBOARD_ARTIFACT = "registry/resource";
+    private static String srcCarDir = null;
+    private static String destCarDir = null;
+    private static String tempDir ;
+
+    public static void main(String arg[]) {
+        BasicConfigurator.configure();
+        DSCarFileMigrationTool dsMigrationTool = new DSCarFileMigrationTool();
+        srcCarDir = arg[0];
+        destCarDir = arg[1];
+        tempDir = srcCarDir.substring(0,srcCarDir.lastIndexOf(File.separator)) +File.separator+"Temp";
```

Comment:
Formatting issue, need a space before and after  + . Mention "Temp" as constant.